src/OpenFeature/ProviderRepository.cs

```diff
@@ -134,18 +135,26 @@ private static async Task InitProviderAsync(
     /// initialization
     /// </param>
     /// <param name="cancellationToken">The <see cref="CancellationToken"/> to cancel any async side effects.</param>
-    public async Task SetProviderAsync(string? domain,
+    internal async Task SetProviderAsync(string? domain,
         FeatureProvider? featureProvider,
         EvaluationContext context,
         Func<FeatureProvider, Task>? afterInitSuccess = null,
         Func<FeatureProvider, Exception, Task>? afterInitError = null,
         CancellationToken cancellationToken = default)
     {
         // Cannot set a provider for a null domain.
+#if NETFRAMEWORK || NETSTANDARD
```

Comment:
We should likely make these functionally the same as they are different this way.  And also update comment if we also want to disallow empty/white space.

Reply:
Actually, did a better simplification here. The only place where `SetProviderAsync` with a domain is called, it is in the `Api` which has already a `null` verification. This cleans up the code.